storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -5486,59 +5486,141 @@ static std::string read_file_as_string(const std::string file) {
   fclose(f);
```

Comment:
add an assertion that size of the file should be less than FN_REFLEN.
If file size is > FN_REFLEN, on release build, print a warning and print the contents to  stderr?